src/groups/mwc/mwcu/mwcu_operationchain.cpp

```diff
@@ -69,17 +62,18 @@ void OperationChain::onOperationCompleted(JobHandle handle)
     // decrement the number of links in this chain and continue executing
     // operations, given the chain is started and is not empty
     if (--d_numLinks != 0 && d_isStarted) {
-        run();
-        return;  // RETURN
+        run(&lock);  // UNLOCK
```

Comment:
I'm not sure passing a `LockGuard` as pointer is very common, given that by definition a `LockGuard` is meant to provide scope based semantic, and a pointer is a different paradigm.

How about simply doing 
`lock.release();` before invoking `run` and letting `run` do `d_mutex.unlock();` ?

Reply:
The way I see it, the guard is meant to provide scoped protection until explicitly released. Usually the releasing is done in the same scope the guard was created, but there is nothing wrong with delegating the release operation to another agent (in this case the `run` function), hence the passing by pointer, as per BDE codestyle rules for modifiable arguments.

Doing the release before calling `run` is technically not exception-safe (although `run` is currently a `noexcept` function, but that might change). This is also doesn't feel "transactional", in the sense that releasing the guard and unlocking the mutex are decoupled, which is not something I would like to do.